### PR TITLE
Elad3d fast test

### PR DESCRIPTION
**Pull request by bolverk**
Fixed clang warnings, cppcheck warnings and added documentation

### CODE REVIEW

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 1]
```diff
@@ -10,534 +10,541 @@
...
+  tetras_[tetra1].neighbors[1] = Nloc;
+  tetras_[tetra1].neighbors[2] = tetra0;
 
-	if(flat_check)
+  if(flat_check)
+    {
+      for (size_t i = 0; i < 4; ++i)
+	b4_temp_[i] = points_[tetras_[Nloc].points[i]];
+      double orient = orient3d(b4_temp_);
+      if (close2zero(orient))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:287]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 2]
```diff
@@ -10,534 +10,541 @@
...
-				tetras_[tetra1].points[1] = temp;
-				temp = tetras_[tetra1].neighbors[0];
-				tetras_[tetra1].neighbors[0] = tetras_[tetra1].neighbors[1];
-				tetras_[tetra1].neighbors[1] = temp;
-			}
-		}
+      for (size_t i = 0; i < 4; ++i)
+	b4_temp_[i] = points_[tetras_[tetra0].points[i]];
+      orient = orient3d(b4_temp_);
+      if (close2zero(orient))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:313]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 3]
```diff
@@ -10,534 +10,541 @@
...
+	      tetras_[tetra0].points[1] = temp;
+	      temp = tetras_[tetra0].neighbors[0];
+	      tetras_[tetra0].neighbors[0] = tetras_[tetra0].neighbors[1];
+	      tetras_[tetra0].neighbors[1] = temp;
+	    }
 	}
+      for (size_t i = 0; i < 4; ++i)
+	b4_temp_[i] = points_[tetras_[tetra1].points[i]];
+      orient = orient3d(b4_temp_);
+      if (close2zero(orient))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:338]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 4]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
+	  break;
 	}
+    }
 	
-	b4_temp_[0] = b3_temp_[1];
-	b4_temp_[1] = b3_temp_[2];
-	b4_temp_[2] = points_[p];
-	b4_temp_[3] = b3_temp_[0];
-	double test0 = orient3d(b4_temp_);
-	if (std::abs(test0) == 0)
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:672]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 5]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
-	{
-		out_check[0] = 1;
-		++out_counter;
-	}	
+  b4_temp_[0] = b3_temp_[1];
+  b4_temp_[1] = b3_temp_[2];
+  b4_temp_[2] = points_[p];
+  b4_temp_[3] = b3_temp_[0];
+  double test0 = orient3d(b4_temp_);
+  if (close2zero(test0))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:679]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 6]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
+    in_counter++;
+  b4_temp_[3] = points_[other_point];
+  double test1 = orient3d(b4_temp_);
+  if (close2zero(test1))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:683]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 7]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
-	{
-		out_check[1] = 1;
-		++out_counter;
-	}
+  b4_temp_[0] = b3_temp_[2];
+  b4_temp_[1] = b3_temp_[0];
+  b4_temp_[2] = points_[p];
+  b4_temp_[3] = b3_temp_[1];
+  test0 = orient3d(b4_temp_);
+  if (close2zero(test0))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:698]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 8]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
+    in_counter++;
+  b4_temp_[3] = points_[other_point];
+  test1 = orient3d(b4_temp_);
+  if (close2zero(test1))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:702]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 9]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
-	{
-		out_check[2] = 1;
-		++out_counter;
-	}
+  b4_temp_[0] = b3_temp_[0];
+  b4_temp_[1] = b3_temp_[1];
+  b4_temp_[2] = points_[p];
+  b4_temp_[3] = b3_temp_[2];
+  test0 = orient3d(b4_temp_);
+  if (std::abs(test0) <1e-12)
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/Delaunay3D.cpp [hunk 10]
```diff
@@ -546,486 +553,486 @@ Delaunay3D::~Delaunay3D()
...
+    in_counter++;
+  b4_temp_[3] = points_[other_point];
+  test1 = orient3d(b4_temp_);
+  if (close2zero(test1))
```
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp:722]: this can make code crash, has to be zero exactly for this to work

File: source/3D/GeometryCommon/HilbertOrder3D.cpp [hunk 11]
```diff
@@ -276,14 +281,14 @@ int HilbertCurve3D::GetRotation(int * piRotation, int iRotationIndex)
...
-		break;
+		//		break;
 	}
 }
 
 // Rotate a shape:
 void HilbertCurve3D::RotateShape(int iShapeIndex, vector<int> vAxes)
 {
-	int iSign = 0;
+  int iSign;
```
Comment (eladtan) [inline source/3D/GeometryCommon/HilbertOrder3D.cpp:291]: why not give initial value?

File: source/3D/GeometryCommon/HilbertOrder3D.cpp [hunk 12]
```diff
@@ -295,28 +300,28 @@ void HilbertCurve3D::RotateShape(int iShapeIndex, vector<int> vAxes)
...
 		// Round off the results:
-		m_vRotatedShapes[iShapeIndex].m_vShapePoints[ii].Round();
+		m_vRotatedShapes[static_cast<size_t>(iShapeIndex)].m_vShapePoints[ii].Round();
 	}
 }
 
 void HilbertCurve3D::RotateShape(HilbertCurve3D_shape const & roShape, HilbertCurve3D_shape & roShapeOut , int iRotationIndex)
 {
-	int iSign = 0;
+  int iSign;
```
Comment (eladtan) [inline source/3D/GeometryCommon/HilbertOrder3D.cpp:322]: why not give default value?

File: source/3D/GeometryCommon/HilbertOrder3D_Utils.cpp [hunk 13]
```diff
@@ -30,9 +31,9 @@ void AdjustPoints(vector<Vector3D> const & vPointsIn, vector<Vector3D> & vPoints
 	double dbScaleY = dbMaxY - dbMinY;
 	double dbScaleZ = dbMaxZ - dbMinZ;
 	// To prevent division by zero (very unlikely - double precision!)
-	bool bFlagX = dbScaleX == 0;
-	bool bFlagY = dbScaleY == 0;
-	bool bFlagZ = dbScaleZ == 0;
+	bool bFlagX = close2zero(dbScaleX);
```
Comment (eladtan) [inline source/3D/GeometryCommon/HilbertOrder3D_Utils.cpp:34]: this is not a good function since it doesn't take into account the magnitude of the points

File: source/3D/GeometryCommon/Voronoi3D.hpp [hunk 14]
```diff
@@ -1,11 +1,11 @@
...
-\brief A 3D Voronoi
-\Author Elad Steinberg
+   \brief A 3D Voronoi
+   \Author Elad Steinberg
 */
 #ifndef VORONOI3D_HPP
 #define VORONOI3D_HPP 1
 
-#define _USE_MATH_DEFINES
+//#define _USE_MATH_DEFINES
```
Comment (eladtan) [inline source/3D/GeometryCommon/Voronoi3D.hpp]: why are you removing this?

File: source/misc/utils.cpp [hunk 15]
```diff
@@ -8,6 +8,12 @@
 #include <cfloat>
 #include <cmath>
 
+bool close2zero(const double x)
```
Comment (eladtan) [inline source/misc/utils.cpp:11]: this is not a good function and should not be used

File: source/misc/utils.cpp [hunk 16]
```diff
@@ -8,6 +8,12 @@
...
+{
+  constexpr double lowest_double = std::numeric_limits<double>::lowest();
+  return std::abs(x)<lowest_double;
```
Comment (eladtan) [inline source/misc/utils.cpp:14]: This is wrong since lowest will give you a huge negative number

File: source/misc/utils.hpp [hunk 17]
```diff
@@ -155,7 +162,8 @@ T LinearInterpolation(const vector<T> &x, const vector<T> &y, T xi)
 	typename vector<T>::const_iterator it = upper_bound(x.begin(), x.end(), xi);
 	if (it == x.end())
 	{
-		if (x.back() == xi)
+	  //		if (x.back() == xi)
```
Comment (eladtan) [inline source/misc/utils.hpp:165]: There is no reason to use this function here

File: source/misc/utils.hpp [hunk 18]
```diff
@@ -168,75 +176,13 @@ T LinearInterpolation(const vector<T> &x, const vector<T> &y, T xi)
...
-/*!
-\brief BiLinear Interpolation
-\param x The x vector, assumed sorted
-\param y The y vector, assumed sorted
-\param z z=f(x,y) matrix, first index is in x
-\param xi The x interpolation location
-\param yi The y interpolation location
-\return f(xi)
-*/
-template <typename T>
```
Comment (eladtan) [inline source/misc/utils.hpp:187]: why is this function removed?

File: source/newtonian/common/LagrangianHLLC.cpp [hunk 19]
```diff
@@ -129,30 +100,23 @@ Conserved LagrangianHLLC::operator()
...
 		energy = (ScalarProd(local_right.Velocity, xdir)*local_right.Energy + ws.right*local_right.Energy*
 			((ws.right - local_right.Velocity.x) / (ws.right - ws.center) - 1))*local_right.Density;
 	}
-	else if (ws.right<0)
+	else
 	{
 		f_gr = fr;
 		energy = ScalarProd(local_right.Velocity, xdir)*local_right.Energy*local_right.Density;
 	}
-	else
```
Comment (eladtan) [inline source/newtonian/common/LagrangianHLLC.cpp:149]: why did you remove the throw?

File: source/newtonian/common/LagrangianHLLC3D.cpp [hunk 20]
```diff
@@ -239,14 +226,12 @@ Conserved3D LagrangianHLLC3D::operator()(ComputationalCell3D const& left, Comput
...
 		if (ws2.left > 0)
 			f_gr = fl;
-		else if (ws2.left <= 0 && ws2.center >= 0)
+		else if (ws2.center >= 0)
 			f_gr = fl + ws2.left*(usl - ul);
-		else if (ws2.center < 0 && ws2.right >= 0)
+		else if (ws2.right >= 0)
 			f_gr = fr + ws2.right*(usr - ur);
-		else if (ws2.right < 0)
-			f_gr = fr;
```
Comment (eladtan) [inline source/newtonian/common/LagrangianHLLC3D.cpp:247]: why did you remove the throw?

File: source/newtonian/common/hllc.cpp [hunk 21]
```diff
@@ -122,19 +93,12 @@ Conserved Hllc::operator()
...
+  else if(ws.center>=0)
    f_gr = fl+ws.left*(usl-ul);
-  else if(ws.center<0&&ws.right>=0)
+  else if(ws.right>=0)
     f_gr  = fr + ws.right*(usr-ur);
-  else if(ws.right<0)
-    f_gr = fr;
   else
-    throw invalid_wave_speeds(local_left,
-			      local_right,
```
Comment (eladtan) [inline source/newtonian/common/hllc.cpp:133]: why no throw?

File: source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp [hunk 22]
```diff
@@ -1,4 +1,4 @@
-#define _USE_MATH_DEFINES
```
Comment (eladtan) [inline source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp:1]: why did you remove this?

File: source/newtonian/three_dimensional/AMR3D.cpp [hunk 23]
```diff
@@ -1095,14 +1108,14 @@ void AMR3D::operator() (HDSim3D &sim)
 	// Update cells
 	ComputationalCell3D cdummy;
 	MPI_exchange_data(tess, cells, true,&cdummy);
-#endif
+	//#endif
 	// Update Max ID
-	size_t & MaxID = sim.GetMaxID();
```
Comment (eladtan) [inline source/newtonian/three_dimensional/AMR3D.cpp:1100]: Why did you comment out this section? It is needed

File: source/newtonian/three_dimensional/Hllc3D_energy.cpp [hunk 24]
```diff
@@ -249,25 +230,23 @@ Conserved3D Hllc3DEnergy::operator()(ComputationalCell3D const& left, Computatio
 		const Conserved3D fl = PrimitiveToFlux(local_left);
 		f_gr = fl;
 	}
-	else if (ws.left <= 0 && ws.center >= 0)
```
Comment (eladtan) [inline source/newtonian/three_dimensional/Hllc3D_energy.cpp:252]: This also protects when the wave speeds have the worng order do don't change it

File: source/newtonian/three_dimensional/Hllc3D_energy.cpp [hunk 25]
```diff
@@ -249,25 +230,23 @@ Conserved3D Hllc3DEnergy::operator()(ComputationalCell3D const& left, Computatio
...
+	else if (ws.center >= 0)
 	{
 		const Conserved3D fl = PrimitiveToFlux(local_left);
 		const Conserved3D usl = starred_state(local_left, ws.left, ws.center);
 		f_gr = fl + ws.left*(usl - ul);
 	}
-	else if (ws.center < 0 && ws.right >= 0)
```
Comment (eladtan) [inline source/newtonian/three_dimensional/Hllc3D_energy.cpp:258]: same comment as above

File: source/newtonian/three_dimensional/Hllc3D_energy.cpp [hunk 26]
```diff
@@ -249,25 +230,23 @@ Conserved3D Hllc3DEnergy::operator()(ComputationalCell3D const& left, Computatio
...
+	else if (ws.right >= 0)
 	{
 		const Conserved3D fr = PrimitiveToFlux(local_right);
 		const Conserved3D usr = starred_state(local_right, ws.right, ws.center);
 		f_gr = fr + ws.right*(usr - ur);
 	}
-	else if (ws.right < 0)
```
Comment (eladtan) [inline source/newtonian/three_dimensional/Hllc3D_energy.cpp:264]: same comment as above

File: source/newtonian/three_dimensional/Hllc3D_energy.cpp [hunk 27]
```diff
@@ -249,25 +230,23 @@ Conserved3D Hllc3DEnergy::operator()(ComputationalCell3D const& left, Computatio
...
+	else
 	{
 		const Conserved3D fr = PrimitiveToFlux(local_right);
 		f_gr = fr;
 	}
-	else
```
Comment (eladtan) [inline source/newtonian/three_dimensional/Hllc3D_energy.cpp:269]: same comment as above

File: source/newtonian/three_dimensional/hdsim_3d.cpp [hunk 28]
```diff
@@ -177,23 +184,23 @@ namespace
 				{
 					res[i] = tess.CalcFaceVelocity(i, point_vel[tess.GetFaceNeighbors(i).first], point_vel[tess.GetFaceNeighbors(i).second]);
 				}
-				catch (UniversalError & eo)
```
Comment (eladtan) [inline source/newtonian/three_dimensional/hdsim_3d.cpp:180]: Keep the eo for clarity

File: source/newtonian/two_dimensional/AreaFix.cpp [hunk 29]
```diff
@@ -591,7 +591,8 @@ vector<Extensive> FluxFix2(Tessellation const& tessold, Tessellation const& tess
 					dA_flux2, eos);
 			std::pair<int, int> mid_neigh(tessmid.GetOriginalIndex(edge2.neighbors.first), tessmid.GetOriginalIndex(edge2.neighbors.second));
 			std::pair<int, int> mid_neigh2(mid_neigh.second, mid_neigh.first);
-			if (only_mid.count(mid_neigh) == 0)
+			//			if (only_mid.count(mid_neigh) == 0)
+			if(only_mid.insert(mid_neigh).second)
 			{
 				only_mid.insert(mid_neigh);
```
Comment (eladtan) [inline source/newtonian/two_dimensional/AreaFix.cpp:597]: This line is no longer needed

File: source/tessellation/VoronoiMesh.cpp [hunk 30]
```diff
@@ -998,7 +1002,7 @@ void VoronoiMesh::GetToTest(vector<vector<int> > &copied,vector<vector<int> > &t
 	for(int i=0;i<nsides;++i)
 		sort(copied[static_cast<size_t>(i)].begin(),copied[static_cast<size_t>(i)].end());
 	totest.resize(static_cast<size_t>(nsides));
-	int test=0;
```
Comment (eladtan) [inline source/tessellation/VoronoiMesh.cpp:1001]: it is good practice to give an initial value

File: source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp [hunk 31]
```diff
@@ -0,0 +1,115 @@
...
+	 write_std_vector_to_hdf5
+	   (geometry,
+	    buffer,
+	    f.first);
+       });
+  }
+
+  // Hydrodynamics
+  {
+    typedef function<double(const ComputationalCell3D&)> my_func;
```
Comment (eladtan) [inline source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: what about interanl energy?

Comment (eladtan) [reply source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: what about tracers and stickers?

Comment (eladtan) [reply source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: There is no mpi data. Is this supposed to replace my write function?

File: source/3D/GeometryCommon/Delaunay3D.cpp (hunk 1 above)
Comment (bolverk) [reply source/3D/GeometryCommon/Delaunay3D.cpp:287]: In the words of clang++: comparing floating point with == or != is unsafe
https://stackoverflow.com/questions/17333/what-is-the-most-effective-way-for-float-and-double-comparison

File: source/3D/GeometryCommon/Delaunay3D.cpp (hunk 2 above)
Comment (bolverk) [reply source/3D/GeometryCommon/Delaunay3D.cpp:313]: See comment above

File: source/3D/GeometryCommon/Delaunay3D.cpp (hunk 3 above)
Comment (bolverk) [reply source/3D/GeometryCommon/Delaunay3D.cpp:338]: ditto

File: source/3D/GeometryCommon/HilbertOrder3D.cpp (hunk 11 above)
Comment (bolverk) [reply source/3D/GeometryCommon/HilbertOrder3D.cpp:291]: Because it is never used

File: source/3D/GeometryCommon/HilbertOrder3D.cpp (hunk 12 above)
Comment (bolverk) [reply source/3D/GeometryCommon/HilbertOrder3D.cpp:322]: Because it is never used

File: source/3D/GeometryCommon/Voronoi3D.hpp (hunk 14 above)
Comment (bolverk) [reply source/3D/GeometryCommon/Voronoi3D.hpp]: It is not necessary anymore

File: source/misc/utils.cpp (hunk 16 above)
Comment (bolverk) [reply source/misc/utils.cpp:14]: You are right. I changed it from lowest() to min()

File: source/misc/utils.hpp (hunk 18 above)
Comment (bolverk) [reply source/misc/utils.hpp:187]: It was never used

File: source/newtonian/common/LagrangianHLLC.cpp (hunk 19 above)
Comment (bolverk) [reply source/newtonian/common/LagrangianHLLC.cpp:149]: It's unreachable code

File: source/newtonian/common/LagrangianHLLC3D.cpp (hunk 20 above)
Comment (bolverk) [reply source/newtonian/common/LagrangianHLLC3D.cpp:247]: Unreachable code

File: source/newtonian/three_dimensional/AMR3D.cpp (hunk 23 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/AMR3D.cpp:1100]: If it is needed then why wasn't it used?

File: source/newtonian/three_dimensional/hdsim_3d.cpp (hunk 28 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/hdsim_3d.cpp:180]: Why? It's not used

File: source/tessellation/VoronoiMesh.cpp (hunk 30 above)
Comment (bolverk) [reply source/tessellation/VoronoiMesh.cpp:1001]: If it is used

File: source/3D/GeometryCommon/HilbertOrder3D.cpp (hunk 11 above)
Comment (eladtan) [reply source/3D/GeometryCommon/HilbertOrder3D.cpp:291]: then remove it

File: source/3D/GeometryCommon/HilbertOrder3D.cpp (hunk 12 above)
Comment (eladtan) [reply source/3D/GeometryCommon/HilbertOrder3D.cpp:322]: remove it

File: source/newtonian/three_dimensional/hdsim_3d.cpp (hunk 28 above)
Comment (eladtan) [reply source/newtonian/three_dimensional/hdsim_3d.cpp:180]: it is thrown later on

File: source/tessellation/VoronoiMesh.cpp (hunk 30 above)
Comment (eladtan) [reply source/tessellation/VoronoiMesh.cpp:1001]: if it is an used variable get rid of it

File: source/3D/GeometryCommon/HilbertOrder3D.cpp (hunk 11 above)
Comment (bolverk) [reply source/3D/GeometryCommon/HilbertOrder3D.cpp:291]: The variable is used, not the initial value

File: source/3D/GeometryCommon/HilbertOrder3D_Utils.cpp (hunk 13 above)
Comment (bolverk) [reply source/3D/GeometryCommon/HilbertOrder3D_Utils.cpp:34]: It didn't take the magnitude into account in the previous implementation

File: source/misc/utils.hpp (hunk 17 above)
Comment (bolverk) [reply source/misc/utils.hpp:165]: This is why I removed it

File: source/newtonian/common/hllc.cpp (hunk 21 above)
Comment (bolverk) [reply source/newtonian/common/hllc.cpp:133]: Unreachable code

File: source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp (hunk 22 above)
Comment (bolverk) [reply source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp:1]: Unnecessary

File: source/newtonian/three_dimensional/Hllc3D_energy.cpp (hunk 24 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/Hllc3D_energy.cpp:252]: No it does not

File: source/newtonian/three_dimensional/hdsim_3d.cpp (hunk 28 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/hdsim_3d.cpp:180]: Yes, but it is not necessary to explicitly invoke the thrown object, see
https://www.ibm.com/docs/en/zos/2.2.0?topic=only-rethrowing-exception-c

File: source/newtonian/two_dimensional/AreaFix.cpp (hunk 29 above)
Comment (bolverk) [reply source/newtonian/two_dimensional/AreaFix.cpp:597]: If it is not needed, why was it there in the first place?

File: source/tessellation/VoronoiMesh.cpp (hunk 30 above)
Comment (bolverk) [reply source/tessellation/VoronoiMesh.cpp:1001]: The value is not used, the variable is

File: source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp (hunk 31 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: I didn't know you had a write function. Where is it?

File: source/3D/GeometryCommon/Delaunay3D.cpp (hunk 9 above)
Comment (eladtan) [inline source/3D/GeometryCommon/Delaunay3D.cpp]: change to close2zero

File: source/3D/GeometryCommon/Voronoi3D.hpp (hunk 14 above)
Comment (eladtan) [reply source/3D/GeometryCommon/Voronoi3D.hpp]: it is necassry for visual studio so keep it

File: source/newtonian/common/LagrangianHLLC.cpp (hunk 19 above)
Comment (eladtan) [reply source/newtonian/common/LagrangianHLLC.cpp:149]: it is recahble if the order of the waves is wrong

File: source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp (hunk 22 above)
Comment (eladtan) [reply source/newtonian/one_dimensional/cylindrical_complementary_1d.cpp:1]: this is needed for visual studio

File: source/newtonian/three_dimensional/AMR3D.cpp (hunk 23 above)
Comment (eladtan) [reply source/newtonian/three_dimensional/AMR3D.cpp:1100]: the value is updated later on, no?

File: source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp (hunk 31 above)
Comment (eladtan) [reply source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: huji-rich/source/3D/GeometryCommon/hdf_write.hpp

File: source/3D/GeometryCommon/Delaunay3D.cpp (hunk 9 above)
Comment (bolverk) [reply source/3D/GeometryCommon/Delaunay3D.cpp]: fixed

File: source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp (hunk 31 above)
Comment (bolverk) [reply source/newtonian/three_dimensional/hdf5_diagnostics_3d.cpp]: Ok. My file is redundant then. I'll remove it.
